normandy/control/static/control/js/components/RecipeForm.jsx

```diff
@@ -1,27 +1,93 @@
 import React from 'react'
 import { connect } from 'react-redux'
 import { Link } from 'react-router'
+import { destroy, reduxForm, getValues } from 'redux-form'
+import { _ } from 'underscore'
+
+import apiFetch from '../utils/apiFetch.js';
+import { parseJsonSchema, generateFieldsFromSchema } from '../utils/formSchemaHelpers.js';
+
 import ControlActions from '../actions/ControlActions.js'
-import { reduxForm } from 'redux-form'
 import composeRecipeContainer from './RecipeContainer.jsx'
-
+import ActionForm from './ActionForm.jsx'
 
 class RecipeForm extends React.Component {
   constructor(props) {
     super(props);
     this.submitForm = this.submitForm.bind(this);
+    this.getAvailableActions = this.getAvailableActions.bind(this);
+    this.changeAction = this.changeAction.bind(this);
+
+    this.state = {
+      availableActions: [],
+      selectedAction: null,
+    };
+  }
+
+  getAvailableActions(recipeId) {
+    apiFetch(`/api/v1/action/`)
+      .then(availableActions => {
+        let selectedActionName = (this.props.recipe ? this.props.recipe.action_name : null);
+
+        availableActions.map(action => {
+          parseJsonSchema(action.arguments_schema).then(schema => {
+            action.arguments_schema = schema;
+            action.fields = generateFieldsFromSchema(schema);
+
+            if (selectedActionName === action.name) {
+              this.setState({
+                selectedAction: action
+              });
+            }
+            return action;
+          })
+        });
+
+        this.setState({
+          availableActions
+        });
+    });
```

Comment:
This is super nitty, so feel free to ignore me. With `.then` on a new line, indenting it always results in these double-detents, which seem weird to me (and harder to read).

I usually do this:

```js
fetchTheThing('foo')
.then(foo => {
   foo.bar();
});
```

instead of this

```
fetchTheThing('foo')
  .then(foo => {
    foo.bar();
  });
```

Either way, I think we should standardize on one or the other. Since you're writing most of this code, you should go with whatever you like better. (Can we encode this in eslint?)